File: app/src/main/java/com/andysklyarov/finnotify/presentation/BindAdapters.java
package com.andysklyarov.finnotify.presentation;

import androidx.annotation.DrawableRes;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.databinding.BindingAdapter;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

public class BindAdapters {
    @BindingAdapter("app:background")
    public static void loadImage(SwipeRefreshLayout layout, @DrawableRes int resource) {
        layout.setBackgroundResource(resource);
    }

    @BindingAdapter("app:background")
    public static void loadImage(ConstraintLayout layout, @DrawableRes int resource) {
        layout.setBackgroundResource(resource);
    }
}

File: app/src/main/java/com/andysklyarov/finnotifyfree/presentation/BindAdapters.kt
package com.andysklyarov.finnotifyfree.presentation

import androidx.annotation.DrawableRes
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.databinding.BindingAdapter
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout


@BindingAdapter("app:background")
fun loadImage(layout: SwipeRefreshLayout, @DrawableRes resource: Int) {
    layout.setBackgroundResource(resource)
}

@BindingAdapter("app:background")
fun loadImage(layout: ConstraintLayout, @DrawableRes resource: Int) {
    layout.setBackgroundResource(resource)
}